sdk/core/dma/dma.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "dma_compartment.h"
+#include "platform/ibex/platform-dma.hh"
+#include <concepts>
+#include <cstddef>
+#include <stdint.h>
+#include <utils.hh>
+
+#if __has_include(<platform-dma.hh>)
+#	include <platform-dma.hh>
+#endif
+
+namespace DMA 
+{
+    template<typename T>
+	concept IsDmaDevice = requires(T device, uint32_t *sourceAddress, uint32_t *targetAddress, size_t lengthInBytes,
+                                        uint32_t sourceStrides, uint32_t targetStrides, uint32_t byteSwapAmount)
+	{
+		{
+			device.write_conf(sourceAddress, targetAddress, lengthInBytes)
+			} -> std::same_as<void>;
+		{
+			device.start_dma()
+			} -> std::same_as<void>;
+        {
+			device.reset_dma()
+			} -> std::same_as<void>;
+	};
+
+    template<typename PlatformDMA>
+
+	requires IsDmaDevice<PlatformDMA>
+
+	class GenericDMA : public PlatformDMA
```

Comment:
I’m a bit confused. Is this a generic DMA interface or a wrapper around a non-CHERI DMA interface in a compartment?